SqlTrainer.UserService.Application/BusinessLogics/IUserBusinessLogic.cs

```diff
@@ -2,4 +2,5 @@
 
 public interface IUserBusinessLogic : IUserService
 {
+    Task<IOperationResult<User>> GetByLoginAsync(string login);
```

Comment:
Do you really need it in UserBusinessLogic?
You can use UserRepository in RegisterBusinessLogic

Reply:
I mean BusinessLogics should use repositories, not other business logics

Reply:
According to your previous comments you`ve said that we should have logic of adding in business logic(hashing password i mean). So we need just to reuse that code without any problems. And in case that dont inject repository and logic I`ve decided to add this.

Reply:
I see. But, for example, in LoginBusinessLogic, as I know, we use IUserRepository, not BusinessLogic. We will add its business logic, I think, but for other stuff, for example, edit or delete or get by id
So, I mean we already use repository in LoginBusinessLogic and I think it has sense to use repository for RegisterBusinessLogic too

Reply:
and dublicate code of password hashing?

Reply:
I guess it is not good idea